storage/cloud-client/notification_polling.py

```diff
@@ -0,0 +1,169 @@
+    ```
+
+3. Create a Cloud Pub/Sub topic and publish bucket notifications there:
+
+   ```
+   $ gsutil notification create -f json -t testtopic gs://testbucket
+   ```
+
+4. Create a subscription for your new topic:
+
+   ```
+   $ gcloud beta pubsub subscriptions create testsubscription --topic=testtopic
+   ```
+
+5. Run this program:
+
+   ```
+   $ python notification_polling testsubscription`
+   ```
+
+6. While the program is running, upload and delete some files in the testbucket
+   bucket (you could use the console or gsutil) and watch as changes scroll by
+   in the app.
+"""
+
+import argparse
+import json
+import sys
+
+from cachetools import LRUCache
+from google.cloud import pubsub
+
+
+ACTION_STRINGS = {
+    'OBJECT_ARCHIVE': 'Object archived',
+    'OBJECT_DELETE': 'Object deleted',
+    'OBJECT_FINALIZE': 'Object created',
+    'OBJECT_METADATA_UPDATE': 'Object metadata updated'
+}
+
+
+DUPLICATE_EVENT_CACHE = LRUCache(10000)
+DUPLICATION_KEY = '{event_type}|{resource}|{metageneration}'
+
+
+def Summarize(message):
+    # [START parse_message]
+    data = message.data
+    attributes = message.attributes
+
+    # The kind of event that just happened. Example: OBJECT_FINALIZE
+    event_type = attributes['eventType']
+    event_description = ACTION_STRINGS.get(
+        event_type, 'Unknown Event %s' % event_type)
+
+    # ID of the bucket. Example: "my_photos"
+    bucket_id = attributes['bucketId']
+    # The ID of the affected objet. Example: "image.jpeg"
+    object_id = attributes['objectId']
+    # Generation of the object. Example: 1234567
+    generation = attributes['objectGeneration']
+    # Format of the attached payload. Example: JSON_API_V1
+    payload_format = attributes['payloadFormat']
+    description = (
+        '{summary} - {bucket_id}/{object_id}\n'
+        '\tGeneration: {generation}\n').format(
+            summary=event_description,
+            bucket_id=bucket_id,
+            object_id=object_id,
+            generation=generation)
+
+    if payload_format == 'JSON_API_V1':
+        # The payload is the JSON API object resource.
+        object_metadata = json.loads(data)
+        size = object_metadata['size']
+        content_type = object_metadata['contentType']
+        metageneration = object_metadata['metageneration']
+        description += (
+            '\tContent type: {content_type}\n'
+            '\tSize: {object_size}\n'
+            '\tMetageneration: {metageneration}\n').format(
+                content_type=content_type,
+                object_size=size,
+                metageneration=metageneration)
+    elif payload_format == 'NONE':
+        # There is no payload.
+        pass
+    else:
+        print('Unknown payload format %s', payload_format)
+    return description
+    # [END parse_message]
+
+
+def GetDedupString(message):
+    """Returns a string unique to this specific event."""
+    # GCS will publish each notification at least once to a Cloud Pub/Sub
```

Comment:
I understand the need for this, but I'm unsure if we want to demonstrate it in this sample. How often do duplicate notifications occur? How important is it to show how to do this in this sample instead of just saying that it indeed needs to be handled? How would you feel about making a separate sample to deal with de-duplication?